crawler/src/main/java/com/brandwatch/minibcr/crawler/service/reddit/auth/RedditAuthenticator.java

```diff
@@ -0,0 +1,46 @@
+package com.brandwatch.minibcr.crawler.service.reddit.auth;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import com.brandwatch.minibcr.crawler.config.RestConfig;
+
+
+@Service
+public class RedditAuthenticator {
+
+    private static final String URL = "https://www.reddit.com/api/v1/access_token";
+
+    private final RestTemplate restTemplate;
+
+    public RedditAuthenticator(RestConfig config) {
+        this.restTemplate = config.restTemplate();
+    }
+
+    public String authenticate(RedditClient client) {
+        return getAuthToken(client);
+    }
+
+    private String getAuthToken(RedditClient client) {
```

Comment:
What's the point of this private method if it's the only method that is called from `authenticate`?